From khali@linux-fr.org Sun Sep  3 13:38:57 2006
Date: Sun, 3 Sep 2006 22:38:52 +0200
From: Jean Delvare <khali@linux-fr.org>
To: Greg KH <greg@kroah.com>
Cc: Linux I2C <i2c@lm-sensors.org>, David Brownell <david-b@pacbell.net>
Subject: [PATCH 11/13] i2c: Constify i2c_algorithm declarations, part 1
Message-Id: <20060903223852.93fb2d4a.khali@linux-fr.org>
Content-Disposition: inline; filename=i2c-algos-constify-i2c_algorithm.patch

i2c: Constify i2c_algorithm declarations, part 1

Make struct i2c_algorithm declarations const in all i2c algorithm
drivers.

Signed-off-by: Jean Delvare <khali@linux-fr.org>
Cc: David Brownell <david-b@pacbell.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/i2c/algos/i2c-algo-bit.c |    2 +-
 drivers/i2c/algos/i2c-algo-pca.c |    2 +-
 drivers/i2c/algos/i2c-algo-pcf.c |    2 +-
 drivers/i2c/algos/i2c-algo-sgi.c |    2 +-
 4 files changed, 4 insertions(+), 4 deletions(-)

--- gregkh-2.6.orig/drivers/i2c/algos/i2c-algo-bit.c
+++ gregkh-2.6/drivers/i2c/algos/i2c-algo-bit.c
@@ -512,7 +512,7 @@ static u32 bit_func(struct i2c_adapter *
 
 /* -----exported algorithm data: -------------------------------------	*/
 
-static struct i2c_algorithm i2c_bit_algo = {
+static const struct i2c_algorithm i2c_bit_algo = {
 	.master_xfer	= bit_xfer,
 	.functionality	= bit_func,
 };
--- gregkh-2.6.orig/drivers/i2c/algos/i2c-algo-pca.c
+++ gregkh-2.6/drivers/i2c/algos/i2c-algo-pca.c
@@ -355,7 +355,7 @@ static int pca_init(struct i2c_algo_pca_
 	return 0;
 }
 
-static struct i2c_algorithm pca_algo = {
+static const struct i2c_algorithm pca_algo = {
 	.master_xfer	= pca_xfer,
 	.functionality	= pca_func,
 };
--- gregkh-2.6.orig/drivers/i2c/algos/i2c-algo-pcf.c
+++ gregkh-2.6/drivers/i2c/algos/i2c-algo-pcf.c
@@ -458,7 +458,7 @@ static u32 pcf_func(struct i2c_adapter *
 
 /* -----exported algorithm data: -------------------------------------	*/
 
-static struct i2c_algorithm pcf_algo = {
+static const struct i2c_algorithm pcf_algo = {
 	.master_xfer	= pcf_xfer,
 	.functionality	= pcf_func,
 };
--- gregkh-2.6.orig/drivers/i2c/algos/i2c-algo-sgi.c
+++ gregkh-2.6/drivers/i2c/algos/i2c-algo-sgi.c
@@ -157,7 +157,7 @@ static u32 sgi_func(struct i2c_adapter *
 	return I2C_FUNC_SMBUS_EMUL;
 }
 
-static struct i2c_algorithm sgi_algo = {
+static const struct i2c_algorithm sgi_algo = {
 	.master_xfer	= sgi_xfer,
 	.functionality	= sgi_func,
 };
